clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -565,8 +566,8 @@ private void recordRebalanceFailure() {
 
         // Note that we override the request timeout using the rebalance timeout since that is the
         // maximum time that it may block on the coordinator. We add an extra 5 seconds for small delays.
-
-        int joinGroupTimeoutMs = Math.max(rebalanceConfig.rebalanceTimeoutMs, rebalanceConfig.rebalanceTimeoutMs + 5000);
+        int joinGroupTimeoutMs = Math.max(client.defaultRequestTimeoutMs(),
```

Comment:
The previous max check was wrong, but an alternative here is to use rebalanceTimeout + 5s in all cases regardless of the request timeout.

Reply:
Could we log a debug info here for the timeout we used?

Reply:
I added the request timeout to the send message in `NetworkClient`. Also made some tweaks for more consistent logging.